cloud/deployment/workerqueue/workerqueue.go

```diff
@@ -35,264 +35,329 @@ var (
+			MinWorkerCount:    wQueueMin,         // use the value from the user input
+			MaxWorkerCount:    wQueueMax,         // use the value from the user input
+			WorkerConcurrency: wQueueConcurrency, // use the value from the user input
+		}
+		fmt.Println(requestedDeployment)
+		// create listToCreate
+		if requestedDeployment.WorkerQueues != nil {
+			fmt.Println(requestedDeployment)
+			fmt.Println(requestedDeployment.WorkerQueues)
+			for i := range *requestedDeployment.WorkerQueues {
+				queues := *requestedDeployment.WorkerQueues
+				existingQueueRequest := astroplatformcore.WorkerQueueRequest{
+					Name:              queues[i].Name,
+					Id:                &queues[i].Id,
+					IsDefault:         queues[i].IsDefault,
+					MaxWorkerCount:    queues[i].MaxWorkerCount,
+					MinWorkerCount:    queues[i].MinWorkerCount,
+					WorkerConcurrency: queues[i].WorkerConcurrency,
+					AstroMachine:      astroplatformcore.WorkerQueueRequestAstroMachine(*queues[i].AstroMachine),
+				}
+				listToCreate = append(listToCreate, existingQueueRequest)
+			}
+		}
+		if name == "" {
+			queueToCreateOrUpdate.Name, err = getQueueName(name, action, &requestedDeployment, out)
+			if err != nil {
+				return err
+			}
+			name = queueToCreateOrUpdate.Name
 		}
+		queueToCreateOrUpdate = SetWorkerQueueValues(wQueueMin, wQueueMax, wQueueConcurrency, queueToCreateOrUpdate, defaultOptions)
 	} else {
 		// get the node poolID to use
-		nodePoolID, err = selectNodePool(workerType, requestedDeployment.Cluster.NodePools, out)
+		cluster, err := deployment.CoreGetCluster("", *requestedDeployment.ClusterId, platformCoreClient)
 		if err != nil {
 			return err
 		}
-	}
-
-	queueToCreateOrUpdate = &astro.WorkerQueue{
-		Name:              name,
-		IsDefault:         false, // cannot create a default queue
-		AstroMachine:      astroMachine.Type,
-		NodePoolID:        nodePoolID,
-		MinWorkerCount:    wQueueMin,         // use the value from the user input
-		MaxWorkerCount:    wQueueMax,         // use the value from the user input
-		WorkerConcurrency: wQueueConcurrency, // use the value from the user input
-	}
-
-	if name == "" {
-		queueToCreateOrUpdate.Name, err = getQueueName(name, action, &requestedDeployment, out)
+		nodePoolID, err = selectNodePool(workerType, *cluster.NodePools, out)
 		if err != nil {
 			return err
 		}
+		queueToCreateOrUpdateHybrid = &astroplatformcore.HybridWorkerQueueRequest{
+			Name:              name,
+			IsDefault:         false, // cannot create a default queue
+			NodePoolId:        nodePoolID,
+			MinWorkerCount:    wQueueMin,         // use the value from the user input
+			MaxWorkerCount:    wQueueMax,         // use the value from the user input
+			WorkerConcurrency: wQueueConcurrency, // use the value from the user input
+		}
+		// create hybridListToCreate
+		for i := range *requestedDeployment.WorkerQueues {
+			queues := *requestedDeployment.WorkerQueues
```

Comment:
same comment, we should move this out of the for loop

Reply:
okay moved